web/firefoxcom.js

```diff
@@ -310,17 +310,67 @@ class FirefoxScripting {
 class MLManager {
   #enabled = new Map();
 
-  constructor({ enableAltText }) {
-    this.#enabled.set("altText", enableAltText);
+  constructor(options) {
+    this.eventBus = null;
+    this.enable({ ...options, listenToProgress: false });
   }
 
-  isEnabledFor(name) {
+  async isEnabledFor(name) {
     return this.#enabled.get(name);
   }
 
   guess(data) {
     return FirefoxCom.requestAsync("mlGuess", data);
   }
+
+  enable({ enableAltText, listenToProgress }) {
+    if (enableAltText) {
+      this.#loadAltTextEngine(listenToProgress);
+    }
+  }
+
+  #loadAltTextEngine(listenToProgress) {
```

Comment:
What happens if this method is invoked *again*, while a previous invocation is still pending?
Do we need to remove any previously added event listeners, or run any other clean-up in that case?